lib/acme/client.rb

```diff
@@ -66,14 +66,19 @@ def connection
     end
   end
 
-  def challenge_from_hash(attributes)
-    case attributes.fetch('type')
+  def challenge_from_hash(arguments)
+    arguments.default_proc = lambda { |_, key| raise "Missing #{key} key" }
```

Comment:
Well I guess, a bit contrived example would be that they have some old hashes after we added another key to the API for some reason and just want to log an error in that case and move on, but still escalate other issues higher.

I think using `Hash#default_proc=` is simply trying to be too smart here tbh, something like this should be good enough:
```rb
def challenge_from_hash(attributes)
  type  = attributes.fetch(:type)
  uri   = attributes.fetch(:uri)
  token = attributes.fetch(:token)

  # ...
end
```
It won't change the error for the initial motivation of the PR but I think we came to the conclusion that there's not much to do about that. If you do want to change the error message nonetheless, one could add `, &KEY_ERROR_PROC)` to each.